src/titiler/extensions/titiler/extensions/wms.py

```diff
@@ -530,8 +535,48 @@ def _reader(src_path: str):
                 return Response(content, media_type=media_type)
 
             elif request_type.lower() == "getfeatureinfo":
-                return Response("Not Implemented", 400)
+                # Required parameters:
+                # - VERSION
+                # - REQUEST=GetFeatureInfo
+                # - LAYERS
+                # - CRS or SRS
+                # - WIDTH
+                # - HEIGHT
+                # - QUERY_LAYERS
+                # - I (Pixel column)
+                # - J (Pixel row)
+                # Optional parameters: INFO_FORMAT, FEATURE_COUNT, ...
 
+                req_keys = {
+                    "version",
+                    "request",
+                    "layers",
+                    "width",
+                    "height",
+                    "query_layers",
+                    "i",
+                    "j",
+                }
+                version, image, format, transparent = get_map_data(req, req_keys)
+                if version == "1.3.0":
+                    i = int(req["i"])
+                    j = int(req["j"])
+
+                    num_bands = image.data.shape[
+                        0
+                    ]  # assuming the image data shape is (num_bands, height, width)
```

Comment:
here we coud directly do 
```
num_bands = image.count
```

https://github.com/cogeotiff/rio-tiler/blob/main/rio_tiler/models.py#L529-L532